### PR TITLE
BUG: scipy.io.hb_write: fails for matrices not in csc format

### PR DESCRIPTION
**Pull request by jor-**
`scipy.io.hb_write` does produce wrong results for matrices not in `csc` format are even fails. This is corrected.

### CODE REVIEW

**Review by perimosocordiae — APPROVED** [review]
I'm surprised the docstrings don't mention anything about requiring CSC format. This is a good, straightforward fix.

CI errors look to be unrelated, so I'm +1 to merge.

**Comment by pv** [comment]
No test

File: scipy/io/harwell_boeing/tests/test_hb.py [hunk 1]
```diff
@@ -58,6 +57,20 @@ def test_simple(self):
         assert_csc_almost_equal(m, SIMPLE_MATRIX)
 
 
+class TestHBReadWrite(object):
+
+    def check_save_load(self, value):
+        with tempfile.NamedTemporaryFile(mode='wt') as file:
+            hb_write(file, value)
+            value_loaded = hb_read(file)
+        assert assert_csc_almost_equal(value, value_loaded)
```
Comment (perimosocordiae) [inline scipy/io/harwell_boeing/tests/test_hb.py]: Looks like an extra `assert` here.

File: scipy/io/harwell_boeing/tests/test_hb.py [hunk 2]
```diff
@@ -58,6 +57,20 @@ def test_simple(self):
         assert_csc_almost_equal(m, SIMPLE_MATRIX)
 
 
+class TestHBReadWrite(object):
```
Comment (perimosocordiae) [inline scipy/io/harwell_boeing/tests/test_hb.py]: Seems like this could be added to `TestRBRoundtrip`, instead of adding a new test class.

**Review by perimosocordiae — CHANGES_REQUESTED** [review]

File: scipy/io/harwell_boeing/tests/test_hb.py (hunk 1 above)
Comment (jor-) [reply scipy/io/harwell_boeing/tests/test_hb.py]: Your right, I remove the assert.

File: scipy/io/harwell_boeing/tests/test_hb.py (hunk 2 above)
Comment (jor-) [reply scipy/io/harwell_boeing/tests/test_hb.py]: I removed `TestRBRoundtrip` and left over only `TestHBReadWrite` because it is a more descriptive name.

**Comment by jor-** [comment]
I have added some tests which have passed successfully.

**Comment by pv** [comment]
thanks, merged

**Comment by jor-** [comment]
thanks

**Pull request merged**